crates/interpreter/src/module.rs

```diff
@@ -199,13 +190,14 @@ impl<'m> Module<'m> {
         unreachable!()
     }
 
-    pub(crate) fn func(&self, x: FuncIdx) -> Parser<'m> {
+    // TODO(dev/fast-interp): Improve the performance of such accessor functions from O(n) to O(1).
+    pub(crate) fn func(&self, x: FuncIdx) -> (Parser<'m>, &[SideTableEntry]) {
```

Comment:
That's probably the problem. The side-table is long-lived.
```suggestion
    pub(crate) fn func(&self, x: FuncIdx) -> (Parser<'m>, &'m [SideTableEntry]) {
```

Reply:
I had tried this, and the error is 
```
error: lifetime may not live long enough
   --> src/module.rs:200:24
    |
105 | impl<'m> Module<'m> {
    |      -- lifetime `'m` defined here
...
194 |     pub(crate) fn func(&self, x: FuncIdx) -> (Parser<'m>, &'m [SideTableEntry]) {
    |                        - let's call the lifetime of this reference `'1`
...
200 |                 return (parser, &self.side_tables[i]);
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method was supposed to return data with lifetime `'m` but it is returning data with lifetime `'1`
```
Then I had tried 
```
  pub(crate) fn func<'a>(&'a self, x: FuncIdx) -> (Parser<'m>, &'m [SideTableEntry])
    where 'a: 'm
```
and the other errors in `exec.rs` remain.

Reply:
Oh yes, I tried to do it and figured out the problem, it's because `Module::side_tables` is still owned by `Module` instead of being borrowed. Ideally we need to implement the TODO, but that's going to be too much work for this PR. I pushed a quick fix to unblock.

Reply:
Thanks!

Reply:
Could you elaborate on why this is a problem in Rust("Module::side_tables is still owned by Module instead of being borrowed")?

Reply:
That's because in Rust you can't borrow a specific field of a struct through a method of that struct without borrowing the whole struct. There are people looking into fixing this. In our particular case, by making the struct (`Module`) borrow its field in the first place, then methods can simply copy that reference instead of borrowing it (thus borrowing the whole struct for the lifetime of that field).